### PR TITLE
[8.0][partner_capital] New module to keep track of public capital information

### PR DESCRIPTION
**Pull request by yajo**
At times it's important to have a classification of your partners depending on their public capital and turnover information.

This module allows such things.

CC @antespi @JavierIniesta @rafaelbn.

### CODE REVIEW

File: partner_capital/views/res_partner_view.xml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    <field name="model">res.partner</field>
+    <field name="inherit_id" ref="base.view_res_partner_filter"/>
+    <field name="arch" type="xml">
+        <filter string="Salesperson" position="after">
+            <filter string="Capital country"
+                    domain="[('is_company', '=', True)]"
+                    context="{'group_by': 'capital_country'}"/>
+            <filter string="Turnover"
+                    domain="[('is_company', '=', True)]"
+                    context="{'group_by': 'turnover_number'}"/>
```
Comment (feketemihai) [inline partner_capital/views/res_partner_view.xml]: I think this is turnover_range_id

File: partner_capital/models/res_partner.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        'res.country',
+        "Capital country",
+        help="Country of origin of this company's capital.")
+    capital_amount = fields.Float(
+        "Capital amount",
+        help="Publicly registered capital amount.")
+    turnover_range_id = fields.Many2one(
+        'res.partner.turnover_range',
+        "Turnover range")
+    turnover_amount = fields.Float()
```
Comment (feketemihai) [inline partner_capital/models/res_partner.py]: A string attribute i think is usefull.

File: partner_capital/views/res_partner_turnover_range_view.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- © 2015 Antiun Ingeniería S.L. - Antonio Espinosa
```
Comment (feketemihai) [inline partner_capital/views/res_partner_turnover_range_view.xml:2]: This are new requirements or why yo put the copywrigths in xml files?

File: partner_capital/README.rst [hunk 4]
```diff
@@ -0,0 +1,82 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+===============
```
Comment (elicoidal) [inline partner_capital/README.rst]: @Yajo, thanks for your contribution!
line until the end of the text (x2)

File: partner_capital/README.rst [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+Partners Capital
+===============
+
+This module extends the functionality of partners to support setting capital
```
Comment (elicoidal) [inline partner_capital/README.rst]: setting the country of origin of capital
(based on your remark, easier to understand and no bis repetita)

File: partner_capital/README.rst [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+partner_capital%0Aversion:%20
+8.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
+
+Credits
+=======
+
+Images
+------
+
+* `Antiun Ingeniería S.L. <http://www.antiun.com>`_: Icon.
```
Comment (elicoidal) [inline partner_capital/README.rst:60]: Remove company icon

File: partner_capital/__init__.py [hunk 7]
```diff
@@ -0,0 +1,7 @@
+# -*- coding: utf-8 -*-
+#    Copyright (c) 2015 Antiun Ingeniería S.L. (http://www.antiun.com)
```
Comment (elicoidal) [inline partner_capital/__init__.py]: harmonize  the 2 lines?

File: partner_capital/models/res_partner.py [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    capital_country = fields.Many2one(
+        'res.country',
+        "Capital country",
+        help="Country of origin of this company's capital.")
+    capital_amount = fields.Float(
```
Comment (elicoidal) [inline partner_capital/models/res_partner.py:17]: I miss a field for currency

File: partner_capital/models/res_partner.py [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+        "Capital amount",
+        help="Publicly registered capital amount.")
+    turnover_range_id = fields.Many2one(
```
Comment (elicoidal) [inline partner_capital/models/res_partner.py:24]: currency

File: partner_capital/views/res_partner_turnover_range_view.xml (hunk 3 above)
Comment (elicoidal) [inline partner_capital/views/res_partner_turnover_range_view.xml:2]: @feketemihai no requirement on OCA guidelines AFAIK.

File: partner_capital/models/res_partner.py (hunk 2 above)
Comment (yajo) [inline partner_capital/models/res_partner.py]: why?

File: partner_capital/views/res_partner_turnover_range_view.xml (hunk 3 above)
Comment (yajo) [inline partner_capital/views/res_partner_turnover_range_view.xml:2]: If AGPL license must be applied on code, I do not care about the language used. Odoo views contain logic written in XML, and I consider it should be copyrighted too. I know it's not required in OCA, but nevermind.

File: partner_capital/README.rst (hunk 6 above)
Comment (yajo) [inline partner_capital/README.rst:60]: It's not our company icon, it's the credits of the icon creator.

Comment (elicoidal) [inline partner_capital/README.rst:60]: Ok: understood!

**Comment by yajo** [comment]
Runbot seems broken.

**Comment by yajo** [comment]
All should work now. Runbot's :x: is https://github.com/odoo/odoo/issues/10159.

**Comment by rafaelbn** [comment]
You put fields in Sales&Purchase but this a general CRM added info and nothing related with Sales&Purchase.

**Comment by JavierIniesta** [comment]
Tested in runbot :+1:

**Comment by JMODaERP** [comment]
I´ve tested in runbot :+1:

**Comment by yajo** [comment]
Let me move the fields outside Sales & Purchases before merging please.

**Comment by yajo** [comment]
@rafaelbn Fixed.

**Comment by rafaelbn** [comment]
OK thanks.

Tested in runbot :+1:

**Pull request merged**